pkg/upgrade/job/job.go

```diff
@@ -137,16 +137,40 @@ func New(plan *upgradeapiv1.Plan, node *corev1.Node, controllerName string) *bat
 						NodeAffinity: &corev1.NodeAffinity{
 							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
 								NodeSelectorTerms: []corev1.NodeSelectorTerm{{
-									MatchExpressions: []corev1.NodeSelectorRequirement{{
-										Key:      corev1.LabelHostname,
-										Operator: corev1.NodeSelectorOpIn,
-										Values: []string{
-											nodeHostname,
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      corev1.LabelHostname,
+											Operator: corev1.NodeSelectorOpIn,
+											Values: []string{
+												nodeHostname,
+											},
 										},
-									}},
+										{
+											Key:      corev1.LabelOSStable,
+											Operator: corev1.NodeSelectorOpIn,
+											Values: []string{
+												"linux",
+											},
+										},
```

Comment:
This would limit upgrade jobs to only be scheduled on linux nodes. Is that intentional? If so, why would we want that? This doesn't match my understanding of this comment by Denise: https://github.com/rancher/rancher/issues/34282#issuecomment-900689190